### PR TITLE
Fix a number of compiler warnings

### PR DESCRIPTION
**Pull request by antonblanchard**

### CODE REVIEW

**Comment by maliberty** [comment]
ready for review?

File: src/cts/src/Clock.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ Box<double> Clock::computeSinkRegionClustered(
   auto yMin = std::numeric_limits<float>::max();
   auto yMax = std::numeric_limits<float>::lowest();
 
-  for (const auto [sinkX, sinkY] : sinks) {
+  for (const auto& [sinkX, sinkY] : sinks) {
```
Comment (rovinski) [inline src/cts/src/Clock.cpp:98]: What's the warning on this one? sinkX and sinkY are floats. They shouldn't need to be references.

**Comment by antonblanchard** [comment]
> ready for review?

Yes. It doesn't remove all warnings with recent compilers, but it reduces it quite a lot.

**Review by maliberty — COMMENTED** [review]
lgtm but please answer @rovinski 's question  (he seems correct)

File: src/cts/src/Clock.cpp (hunk 1 above)
Comment (maliberty) [reply src/cts/src/Clock.cpp:98]: See discussion in https://github.com/The-OpenROAD-Project/OpenROAD/pull/2751

**Review by maliberty — APPROVED** [review]

**Pull request merged**